src/js/common/components/asyncLoader.jsx

```diff
@@ -0,0 +1,30 @@
+// Adapted from https://gist.github.com/acdlite/a68433004f9d6b4cbc83b5cc3990c194
+import React from 'react';
+import LoadingIndicator from '../../common/components/LoadingIndicator';
+
+export default function asyncLoader(getComponent, message) {
+  return class AsyncComponent extends React.Component {
+    static Component = null;
```

Comment:
Actually, I think the use case is for re-visiting a route. It's caching the component that the promise returns so that if you come back to a route it doesn't have to render twice.

That doesn't really matter for us, but I'm going to leave it so this stays generic enough for other uses.